[CI] SpatialPushDownGeoPointIT testPushedDownQueriesSingleValue failing

**Build Scans:**
- [elasticsearch-periodic-platform-support #3444 / ubuntu-1804_platform-support-unix](https://gradle-enterprise.elastic.co/s/k5wrke4dfv4be)
- [elasticsearch-pull-request #26303 / part-3](https://gradle-enterprise.elastic.co/s/zldcag72tjsx2)
- [elasticsearch-pull-request #26234 / part-3](https://gradle-enterprise.elastic.co/s/t2al5rmiqw4ds)
- [elasticsearch-periodic-platform-support #3428 / ubuntu-2004_platform-support-unix](https://gradle-enterprise.elastic.co/s/q455werbsglnm)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:internalClusterTest" --tests "org.elasticsearch.xpack.esql.spatial.SpatialPushDownGeoPointIT.testPushedDownQueriesSingleValue" -Dtests.seed=F76B6D4D0C6ABE1F -Dtests.locale=ja-JP-u-ca-japanese-x-lvariant-JP -Dtests.timezone=America/El_Salvador -Druntime.java=22
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('testPushedDownQueriesSingleValue'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.xpack.esql.spatial.SpatialPushDownGeoPointIT'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))

**Failure Message:**
```
java.lang.AssertionError: expected:<68> but was:<66>
```

**Issue Reasons:**
- [main] 4 failures in test testPushedDownQueriesSingleValue (1.5% fail rate in 264 executions)
- [main] 2 failures in step part-3 (1.8% fail rate in 109 executions)
- [main] 2 failures in pipeline elasticsearch-periodic-platform-support (40.0% fail rate in 5 executions)
- [main] 2 failures in pipeline elasticsearch-pull-request (1.8% fail rate in 109 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -70,9 +70,6 @@ tests:
   issue: https://github.com/elastic/elasticsearch/issues/110801
 - class: org.elasticsearch.nativeaccess.VectorSystemPropertyTests
   method: testSystemPropertyDisabled
-  issue: https://github.com/elastic/elasticsearch/issues/110949
-- class: org.elasticsearch.xpack.esql.spatial.SpatialPushDownGeoPointIT
-  method: testPushedDownQueriesSingleValue
   issue: https://github.com/elastic/elasticsearch/issues/111084
 - class: org.elasticsearch.multi_node.GlobalCheckpointSyncActionIT
   issue: https://github.com/elastic/elasticsearch/issues/111124

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -70,7 +70,7 @@ tests:
   issue: https://github.com/elastic/elasticsearch/issues/110801
 - class: org.elasticsearch.nativeaccess.VectorSystemPropertyTests
   method: testSystemPropertyDisabled
-  issue: https://github.com/elastic/elasticsearch/issues/111084
+  issue: https://github.com/elastic/elasticsearch/issues/110949
 - class: org.elasticsearch.multi_node.GlobalCheckpointSyncActionIT
   issue: https://github.com/elastic/elasticsearch/issues/111124
 - class: org.elasticsearch.cluster.PrevalidateShardPathIT

--- a/x-pack/plugin/esql/src/internalClusterTest/java/org/elasticsearch/xpack/esql/spatial/SpatialPushDownGeoPointIT.java
+++ b/x-pack/plugin/esql/src/internalClusterTest/java/org/elasticsearch/xpack/esql/spatial/SpatialPushDownGeoPointIT.java
@@ -7,10 +7,13 @@
 
 package org.elasticsearch.xpack.esql.spatial;
 
+import org.apache.lucene.geo.GeoEncodingUtils;
 import org.elasticsearch.geo.GeometryTestUtils;
 import org.elasticsearch.geometry.Geometry;
 
 public class SpatialPushDownGeoPointIT extends SpatialPushDownTestCase {
+
+    private static final double LAT_MAX_VALUE = GeoEncodingUtils.decodeLatitude(Integer.MAX_VALUE - 3);
 
     @Override
     protected String fieldType() {
@@ -19,7 +22,9 @@ public class SpatialPushDownGeoPointIT extends SpatialPushDownTestCase {
 
     @Override
     protected Geometry getIndexGeometry() {
-        return GeometryTestUtils.randomPoint();
+        // This is to overcome lucene bug https://github.com/apache/lucene/issues/13703.
+        // Once it is fixed we can remove this workaround.
+        return randomValueOtherThanMany(p -> p.getLat() >= LAT_MAX_VALUE, GeometryTestUtils::randomPoint);
     }
 
     @Override

